bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/dto/SetAttributesRequest.java

```diff
@@ -45,6 +44,6 @@ public SetAttributesRequest(@Nullable String name, @Nullable List<HueObject> lig
         }
 
         this.name = name;
-        this.lights = lights == null ? null : lights.stream().map(l -> l.getId()).collect(Collectors.toList());
+        this.lights = lights == null ? null : lights.stream().map(l -> l.getId()).toList();
```

Comment:
Changing `.collect(Collectors.toList())` (Java 8+ syntax) => to `.toList()` (Java 16+ syntax) means that it is no longer possible to backport the code to OH v3.4.x.

IMHO it is bad practice to change perfectly working code to a newer Java version syntax, when the older version syntax is not deprecated, and continues to work just fine. 

Note: the same comment applies for all the other nine similar cases.

Reply:
This was changed for a reason - namely to fix:
```
Unsafe null type conversion (type annotations): The value of type 'java.util.stream.Collector<java.lang.@NonNull String,capture#of ?,java.util.@NonNull List<java.lang.@NonNull String>>' is made accessible using the less-annotated type 'java.util.stream.@NonNull Collector<? super java.lang.String,java.lang.Object,java.util.@NonNull List<java.lang.@NonNull String>>'
```

You can still backport to previous versions of openHAB if you want, you just need to change back to `collect(Collectors.toList());` in your port, like you also need to maintain `binding.xml` rather than `addon.xml` etc. All these warnings were fixed in the API v1 related code, so there wouldn't be any compelling reason to backport these fixes though.

Reply:
> you just need to 

One of the most dangerous phases in the English language..